src/Runner.Common/RunServer.cs

```diff
@@ -67,10 +67,48 @@ private void CheckConnection()
             }
         }
 
-        public Task<AgentJobRequestMessage> GetJobMessageAsync(string id)
+        public Task<AgentJobRequestMessage> GetJobMessageAsync(string id, CancellationToken cancellationToken)
         {
             CheckConnection();
-            return _taskAgentClient.GetJobMessageAsync(id);
+            var jobMessage = RetryRequest<AgentJobRequestMessage>(async () =>
+                                                    {
+                                                        return await _taskAgentClient.GetJobMessageAsync(id, cancellationToken);
+                                                    }, cancellationToken);
+            return jobMessage;
+        }
+
+        private async Task<T> RetryRequest<T>(Func<Task<T>> func,
+                                                        CancellationToken cancellationToken,
+                                                        int maxRetryAttempts = 5
+                                                        )
+        {
+            var currentRetryAttempt = 1;
```

Comment:
NIT: consider renaming `currentRetryAttempt` to `retryCount`, it is the terminology most often used across the runner code.